src/rfc/rebuild_invocation_args.php

```diff
@@ -0,0 +1,185 @@
+<?hh // strict
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+namespace Facebook\ShipIt;
+
+use namespace HH\Lib\{Str, C, Vec};
+use Facebook\HackCodegen\{
+  HackCodegenFactory,
+  HackCodegenConfig,
+  HackBuilderValues,
+  HackBuilderKeys,
+};
+
+const string ARG_SUFFIX = "::";
+const dict<string, string> CAPITALIZATION_MAP = dict[
+  "ssh" => "SSH",
+  "https" => "HTTPS",
+  "lfs" => "LFS",
+  "github" => "GitHub",
+];
+
+type ShipItFunctionArg = shape(
+  'cli_name' => string,
+  'func_name' => string,
+  'takes_arg' => bool,
+  'description' => ?string,
+);
+
+function getFunctionArg(ShipItCLIArgument $arg): ?ShipItFunctionArg {
```

Comment:
Prefer a final abstract class with public methods over top-level functions (I know it's not as necessary since we have namespaces here, but it's good for consistency).